services/venice-router/src/test/java/com/linkedin/venice/router/TestMetaDataHandler.java

```diff
@@ -185,13 +218,242 @@ public void testControllerLookup() throws IOException {
     Assert.assertEquals(controllerResponse.getUrl(), "http://" + leaderControllerHost + ":" + leaderControllerPort);
   }
 
-  // The deprecated non inclusive URL must also continue to work.
+  // The deprecated non-inclusive URL must also continue to work.
   @DataProvider(name = "controllerUrlProvider")
   public static Object[][] dataProvider() {
     // go/inclusivecode deprecated (alias="leader_controller")
     return new Object[][] { { "master_controller" }, { "leader_controller" } };
   }
 
+  @DataProvider(name = "blobDiscoveryProvider")
+  public static Object[][] blobDiscoveryDataProvider() {
```

Comment:
I am not a big fan of this huge data provider. We use data provider because we want to examine different combination of inputs, for example, [this one](https://github.com/linkedin/venice/blob/5c0e66a203be1e9cfd0c6cad27597035543f06a9/services/venice-router/src/test/java/com/linkedin/venice/router/TestAdminOperationsHandler.java#L194). 

If I understand your intention right, we are just providing the mocked response in this data provider so it's better just to mock them in the test method itself instead of having a data provider

Reply:
Agree. Not a fan of a huge data provider either! 

We have dynamic response based on the inputs.

Another option I was thinking of is having a separate test method but thought this might be easier to read the test case. I can change this